### PR TITLE
Release 2.9.0

### PR DESCRIPTION
**Pull request by d4straub**
This is release 2.9.0.

**Potential problem:**
For a new software I produced a container and deposited it at dockerhub. The processes that use that software are doing so with `container 'docker.io/d4straub/pipesidle:0.1.0-beta'`. My proposal to put the tool (pypi packaged) to conda was rejected by the developer of said tool. I generated the container with [Dockerfile](https://github.com/d4straub/pipesidle/blob/83600d3b4068526e19d8f9db7a1c716c2a7c5c0a/Dockerfile) & [environment.yml](https://github.com/d4straub/pipesidle/blob/83600d3b4068526e19d8f9db7a1c716c2a7c5c0a/environment.yml). Maybe there is a better solution to that?

**Changes are:**
- added multiple regions analysis (including 5R / SMURF / q2-sidle), for that the meta map was extended and several new processes were added
- Template update 2.11 to 2.13.1, that might be the majority of file changes
- Plethora of bug fixes, for details see CHANGELOG.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by erikrikarddaniel — APPROVED** [review]
That's a lot of files. Looks fine.
The container question sounds a bit like an application of "Wave", but I don't know.

File: docs/usage.md [hunk 1]
```diff
@@ -246,6 +247,37 @@ Special features of taxonomic classification tools:
...
+
+For example, multiple variable regions of the 16S rRNA gene were sequenced with various primers and need to be unified. This leads to one unified abundance and taxonomy profile over all variable regions. However, ASV sequences are only available separately, there is no reconstruction of complete de-novo sequences feasible.
+
+Required is information about sequencing data via [`--input`](#samplesheet-input), region primers length information via [`--multiregion`](https://nf-co.re/ampliseq/parameters#multiregion), and a taxonomic database via [`--sidle_ref_taxonomy`](https://nf-co.re/ampliseq/parameters#sidle_ref_taxonomy) or [`--sidle_ref_tax_custom`](https://nf-co.re/ampliseq/parameters#sidle_ref_tax_custom).
```
Comment (mashehu) [inline docs/usage.md:256]:
```suggestion
Information about sequencing data via [`--input`](#samplesheet-input), region primers length information via [`--multiregion`](https://nf-co.re/ampliseq/parameters#multiregion), and a taxonomic database via [`--sidle_ref_taxonomy`](https://nf-co.re/ampliseq/parameters#sidle_ref_taxonomy) or [`--sidle_ref_tax_custom`](https://nf-co.re/ampliseq/parameters#sidle_ref_tax_custom) is required.
```

File: modules/local/sidle_taxrecon.nf [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    def args = task.ext.args ?: ''
+    """
+    #https://q2-sidle.readthedocs.io/en/latest/reconstruction.html#taxonomic-reconstruction
+    #https://forum.qiime2.org/t/sidle-reconstruct-database/25439
+    export XDG_CONFIG_HOME="./xdgconfig"
+    export MPLCONFIGDIR="./mplconfigdir"
+    export NUMBA_CACHE_DIR="./numbacache"
+
+    #CPU=1
```
Comment (mashehu) [inline modules/local/sidle_taxrecon.nf:32]:
```suggestion
```

File: modules/local/sidle_trim.nf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    def primerfw = "${meta.fw_primer}"
+    def primerrv = "${meta.rv_primer}"
+    def length = "${meta.region_length}"
+    """
+    # https://q2-sidle.readthedocs.io/en/latest/read_preparation.html#dada2
+    export XDG_CONFIG_HOME="./xdgconfig"
+    export MPLCONFIGDIR="./mplconfigdir"
+    export NUMBA_CACHE_DIR="./numbacache"
+
+    #CPU=1
```
Comment (mashehu) [inline modules/local/sidle_trim.nf:34]:
```suggestion
```

File: nextflow.config [hunk 4]
```diff
@@ -13,6 +13,7 @@ params {
     input                      = null
     input_fasta                = null
     input_folder               = null
+    multiregion          = null
     extension                  = "/*_R{1,2}_001.fastq.gz"
```
Comment (mashehu) [inline nextflow.config:17]:
```suggestion
    multiregion                = null
    extension                  = "/*_R{1,2}_001.fastq.gz"
```

File: nextflow_schema.json [hunk 5]
```diff
@@ -525,8 +542,31 @@
...
+                },
+                "sidle_ref_tax_custom": {
+                    "type": "string",
+                    "help_text": "Consider also setting `--sidle_ref_tree_custom`. Example usage: `--sidle_ref_tax_custom 'rep_set_99.fasta,rep_set_aligned_99.fasta,taxonomy_99_taxonomy.txt'`",
+                    "description": "Comma separated paths to three files: reference taxonomy sequences (*.fasta), reference taxonomy strings (*.txt)"
+                },
+                "sidle_ref_tree_custom": {
+                    "type": "string",
+                    "help_text": "Overwrites tree chosen by `--sidle_ref_taxonomy`",
+                    "description": "Path to SIDLE reference taxonomy tree (*.qza)"
```
Comment (mashehu) [inline nextflow_schema.json:567]: maybe add `(*.qza)` as a pattern?

**Review by mashehu — APPROVED** [review]

File: nextflow_schema.json (hunk 5 above)
Comment (d4straub) [reply nextflow_schema.json:567]: Will do, thanks!

**Comment by d4straub** [comment]
Unfortunately the AWS full test run failed, will re-try and then troubleshoot.

**Pull request merged**